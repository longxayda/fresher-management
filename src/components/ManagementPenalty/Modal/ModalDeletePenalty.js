import React, { useState } from "react";
import { Button, Modal, Row, Col } from "react-bootstrap";
import { deleteOneBonusPenalty } from "redux/fresherManageSlice/penaltySlice";
import { useDispatch, useSelector } from "react-redux";
import { isProcessingSelector } from "redux/selectors/penaltySelector"

function ModalDeletePenalty({ no, id, type, notifyDelete }) {
  const dispatch = useDispatch();
  const isProcessing = useSelector(isProcessingSelector);

  const [show, setShow] = useState(false);
  const handleClose = () => {
    setShow(false);
  };

  const deleted = async () => {
    try {
      await dispatch(deleteOneBonusPenalty(id)).unwrap();
      notifyDelete("success");
      handleClose();
    }
    catch (err) {
      notifyDelete("failed");
    }
  }
  const handleShow = () => setShow(true);
  const footerDelete = {
    paddingLeft: "20%",
    paddingTop: "16px",
    borderTop: "1px solid #e9ecef",
  };
  const headerDelete = {
    borderBottom: "1px solid #e9ecef",
  };
  const titleDelete = {
    marginTop: "0",
  };

  return (
    <>
      <Button
        id="delete-pelnaty-btn"
        type="button"
        variant="danger"
        className="position-action-delete"
        onClick={handleShow}
        size="sm"
      >
        <i className="fas fa-trash"></i>
      </Button>
      <Modal
        show={show}
        onHide={handleClose}
        centered
      >
        <Modal.Header closeButton style={headerDelete}>
          <Modal.Title style={titleDelete} className="text-danger modal-title h4">
            DELETE {type.toUpperCase()} NO {no}
          </Modal.Title>
          {isProcessing && <div className="processing"></div>}
        </Modal.Header>

        <Modal.Body>
          <h5>
            Are you sure you want to delete {type.toUpperCase()} NO {no}?
          </h5>
        </Modal.Body>

        <Modal.Footer style={footerDelete}>
          <Row>
            <Col lg="2" md sm xs="2"></Col>
            <Col lg="5" md sm xs="5">
              <Button
                id="delete-btn"
                className="btn-fill btn-wd"
                variant="danger"
                onClick={deleted}
              >
                Delete
              </Button>
            </Col>
            <Col lg="2" md sm xs="5">
              <Button
                id="delete-btn"
                className="btn-fill btn-wd"
                variant="secondary"
                onClick={handleClose}
              >
                Close
              </Button>
            </Col>
          </Row>
        </Modal.Footer>
      </Modal>
    </>
  );
}
export default ModalDeletePenalty;