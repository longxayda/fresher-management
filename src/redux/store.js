import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
