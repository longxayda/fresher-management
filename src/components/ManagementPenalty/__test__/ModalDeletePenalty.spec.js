import React from "react";
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import store from "redux/store";
import { Provider } from "react-redux";
import ModalDeletePenalty from "components/ManagementPenalty/Modal/ModalDeletePenalty";

configure({ adapter: new Adapter() });

describe("ModalDeletePenalty", () => {
  let component;
  beforeEach(() => {
    component = mount(<Provider store={store}> <ModalDeletePenalty /> </Provider>);
  });

  it("should render correctly", () => {
    expect(component.html()).toMatchSnapshot();
  });
  
  it("should have correct btn number", () => {
    expect(component.find('Button').length).toBe(1);
  });

  it("should have button", () => {
    expect(component.find('button').exists()).toEqual(true);
  });

  it ("should have modal", () => {
    expect(component.find('#delete-btn').exists()).toEqual(false);
    component.find('#delete-pelnaty-btn').first().simulate('click');
    expect(component.find('#delete-btn').exists()).toEqual(true);
  });
});