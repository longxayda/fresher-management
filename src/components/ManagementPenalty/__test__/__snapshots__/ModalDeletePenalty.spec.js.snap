// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDeletePenalty should render correctly 1`] = `" <button type=\\"button\\" id=\\"delete-pelnaty-btn\\" class=\\"position-action-delete btn btn-danger\\"><i class=\\"fas fa-trash\\"></i></button> "`;
